bin/check-input-datasets.js

```diff
@@ -0,0 +1,153 @@
+const fsPromises = require("fs").promises;
+const Ajv = require("ajv").default;
+
+const { DATA_FOLDER_NAME } = require("../src/process-single-dataset/constants");
+const {
+  readDatasetJson,
+  readAndParseFile,
+  readPossibleZippedFile,
+} = require("../src/utils");
+const dataPrep = require("../src/data-validation/data-prep");
+
+// referenced partial schemas
+const refSchemas = [
+  require("../src/data-validation/schema/definitions.schema.json"),
+  require("../src/data-validation/schema/discrete-feature-options.schema.json"),
+  require("../src/data-validation/schema/array-items/discrete-feature-option.schema.json"),
+  require("../src/data-validation/schema/array-items/file-info.schema.json"),
+  require("../src/data-validation/schema/array-items/input-measured-features.schema.json"),
+];
+
+// Schemas that describe the handoff files that define a dataset or a
+// collection of datasets.
+const inputMegaset = require("../src/data-validation/schema/input-megaset.schema.json");
+const inputDatasetInfo = require("../src/data-validation/schema/input-dataset-info.schema.json");
+const inputMeasuredFeatures = require("../src/data-validation/schema/input-measured-features-doc.schema.json");
+const inputDataSet = require("../src/data-validation/schema/input-dataset.schema.json");
+const inputImages = require("../src/data-validation/schema/images.schema.json");
+const featureDef = require("../src/data-validation/schema/feature-def.schema.json");
+
+const INPUT_DATASET_SCHEMA = [
+  ...refSchemas,
+  inputImages,
+  featureDef,
+  inputMegaset,
+  inputDatasetInfo,
+  inputMeasuredFeatures,
+  inputDataSet,
+];
+const INPUT_DATASET_SCHEMA_FILE = "input-dataset.schema.json";
+const INPUT_MEGASET_SCHEMA_FILE = "input-megaset.schema.json";
+
+const ajv = new Ajv({
+  coerceTypes: true,
+  removeAdditional: true,
+  schemas: INPUT_DATASET_SCHEMA,
+});
```

Comment:
most of lines 12-46 could be refactored into src/ as a single function with a name like getInputDatasetSchema